homeassistant/components/modbus/validators.py

```diff
@@ -385,7 +387,11 @@ def validate_entity(
                     del entities[entity_inx]
                 else:
                     entity_inx += 1
-
+        if no_entities:
+            err = f"Modbus {hub[CONF_NAME]} is instable without entities, please add entities!"
+            _LOGGER.warning(err)
+            # Ensure timeout is not started/handled.
+            hub[CONF_TIMEOUT] = -1
```

Comment:
What would be a use case of adding a hub without adding entities?

Reply:
In a HA core only, none !

But there are a couple of custom components, who use a base modbus configuration, to add there own configuration on top...and thus use the modbus integration code instead of duplicating it (duplicating it gives nasty dependencies because both use the same library).

Reply:
No, it is only when running without entities....otherwise I would have issued an error and stopped the loading.

Reply:
Hang on, I made a mistake....the discussion we are having belong to PR #11618, and should not be here...

this one is just a single "if".

Sorry for the mistake, give me 2 seconds, and you will have a commit that correct this PR.

Reply:
@joostlek updated to be correct....when I created the branch, I forgot to start from dev :-( my mistake.